src/webui/webapplication.cpp

```diff
@@ -774,6 +777,33 @@ void WebApplication::action_command_bottomPrio()
     BitTorrent::Session::instance()->bottomTorrentsPriority(hashes);
 }
 
+void WebApplication::action_command_setLocation()
+{
+    CHECK_URI(0);
+    CHECK_PARAMETERS("hashes" << "location");
+
+    QStringList hashes = request().posts["hashes"].split("|");
+    QString newLocationPath = request().posts["location"].trimmed();
+
+    // check location exists
+    QFileInfo newLocation(newLocationPath);
+    if (!newLocation.exists()) {
+        Logger::instance()->addMessage(tr("Invalid path '%1'").arg(newLocationPath));
+        return;
+    }
+
+    foreach (const QString &hash, hashes) {
+        BitTorrent::TorrentHandle *const torrent = BitTorrent::Session::instance()->findTorrent(hash);
+        if (torrent) {
```

Comment:
Of course it make sense, it filters out invalid entries and saves a pile of redundant indentions.
but it's alright if you insist the current one.